src/bokeh/plotting/geo_feature.py

```diff
@@ -0,0 +1,176 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from functools import wraps
+
+# External imports
+import numpy as np
+
+# Bokeh imports
+from ..util.dependencies import import_optional
+
+# Optional imports
+cartopy = import_optional("cartopy")
+
+
+def requires_cartopy(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if cartopy is None:
+            raise ModuleNotFoundError(
+                f"Missing optional dependency 'cartopy' for {type(f).__name__!r}. "
+                "To use this function, please install 'cartopy'.",
+            )
+        return f(*args, **kwargs)
+    return wrapper
```

Comment:
Can we avoid all this complexity and just raise a useful error immediately at the top of the module if cartopy is not present?